cpp/models/ode_seir/model.h

```diff
@@ -108,8 +108,33 @@ class Model : public CompartmentalModel<InfectionState, Populations<InfectionSta
         return temp;
     }
 
+    /**
+    * @brief interpolate reproduction numbers at freely chosen time value that lies in between the time points of the given time series.
+    * values are linearly interpolated from their immediate neighbors from the old time points.
+    * @param y TimeSeries whose reproduction numbers we want to interpolate
+    * @param t_value double time at which we approximate the reproduction number
+    * @return interpolated reproduction number at given time value
+    */
+    IOResult<double> interpolate_reproduction_numbers(const mio::TimeSeries<ScalarType>& y, const double t_value)
+    {
+        if (t_value < -1 || t_value > y.get_num_time_points()) {
```

Comment:
I would reduce the interval, so no extrapolation is possible. 

Additionaly, what if we start with an `t_0` not equal zero here. I think this should also be adaptive with         y.get_time(0).